checkpoint_schedules/schedule.py

```diff
@@ -19,48 +21,71 @@
 
 
 class StorageType(Enum):
-    """
-    RAM : It is the first type of checkpoint storage.
+    """ Enums to indicate the storage type of the checkpoint data.
+
+    RAM : A enum to indicate the storage of the forward checkpoint data in memory.
 
-    DISK : It is the second type of checkpoint storage.
+    DISK : A enum to indicate the storage of the forward checkpoint data in RAM.
     
-    WORK : ...
+    FWD_RESTART : A enum to indicate the storage of the forward checkpoint data
+    in a local used to initilise the forward solver.
 
-    NONE : Indicate that there is no specific storage location defined 
+    ADJ_DEPS : A enum to indicate the storage of the forward checkpoint data
+    in a local used for the adjoint computation.
+
+    NONE : A enum to indicate that there is no specific storage location defined 
     for the checkpoint data.
     """
     RAM = 0
     DISK = 1
-    WORK = -1
+    FWD_RESTART = 2
+    ADJ_DEPS = 3
     NONE = None
 
 
 class StepType(IntEnum):
-    """Step type cost.
+    """Enums to indicate which execution is being executed in such step.
+
+    FORWARD : A enum to indicate the Forward execution is such step.
+
+    FORWARD_REVERSE : A enum to indicate the foward and adjoint execution in
+    such step.
+
+    WRITE_ADJ_DEPS : A enum to indicate that the storage of the forward data
+    required for the adjoint computation is such step.
+
+    WRITE_ICS : A enum to indicate that the storage of the forward data
+    required for the the forward solver restarting.
+
+    READ_ADJ_DEPS : A enum to indicate that the read of the forward data
+    required for the adjoint computation is such step.
+
+    READ_ICS : A enum to indicate that the read of the forward data required
+    for the the forward solver restarting.
     """
     NONE = 0
     FORWARD = 1
     FORWARD_REVERSE = 2
-    WRITE_DATA = 3
+    WRITE_ADJ_DEPS = 3
     WRITE_ICS = 4
-    READ_DATA = 5
+    READ_ADJ_DEPS = 5
     READ_ICS = 6
 
 
 class CheckpointAction:
-    """A checkpoint action.
-    
+    """A checkpoint action base class.
+
     Attributes
     ----------
     *args : Any
-        The *args correspond to the arguments of the checkpoint scheduled actions: 
-        `Forward`, `Reverse`, `Copy`, `EndForward`, `EndReverse`.
-    
+        The *args correspond to the arguments of the checkpoint scheduled 
+        actions: `Forward`, `Reverse`, `Copy`, `EndForward`, `EndReverse`.
```

Comment:
Missing `Move`. Maybe do not list subclasses, as it's difficult to maintain a list like this.